src/Symfony/Component/BrowserKit/HttpBrowser.php

```diff
@@ -61,7 +61,7 @@ protected function doRequest($request): Response
      */
     private function getBodyAndExtraHeaders(Request $request, array $headers): array
     {
-        if (\in_array($request->getMethod(), ['GET', 'HEAD'])) {
+        if (('GET' === $request->getMethod() && !isset($headers['content-type'])) || 'HEAD' === $request->getMethod()) {
```

Comment:
why only the `GET` method?
Head is about receiving headers in response, not the request.

IMHO, everything I can do with GET should be possible with HEAD (except reading response body)

Reply:
You made me think! So I took a look at the [RFC7231 - 4.3.2](https://tools.ietf.org/html/rfc7231#section-4.3.2) and I believe it fits to what you've said @jderusse .
At the end, a HEAD request only instructs the server to not send a message body in response.
I'll think on a change + tests and come back here with it.

Reply:
Awesome. 
I'll be happy to merge when this is resolved.

Reply:
Code updated and another test added to cover for that